extensions/amp-render/1.0/amp-render.js

```diff
@@ -246,9 +247,38 @@ export class AmpRender extends BaseElement {
       this.mutateProps(
         dict({
           'render': (data) => {
+            if (this.element.getAttribute('binding') === 'no') {
+              return templates
+                .renderTemplateAsString(dev().assertElement(template), data)
+                .then((html) => dict({'__html': html}));
+            }
+
+            let element;
             return templates
-              .renderTemplateAsString(dev().assertElement(template), data)
-              .then((html) => dict({'__html': html}));
+              .renderTemplate(dev().assertElement(template), data)
+              .then((el) => {
+                element = el;
+                return Services.bindForDocOrNull(this.element);
+              })
+              .then((bind) => {
+                const bindingAttrValue = this.element.getAttribute('binding');
+                return bind.rescan([element], [], {
+                  'fast': true,
+                  'update':
+                    // update should be true if:
+                    // 1. no binding is specified (default is binding=always) or
+                    // 2. binding=always is specified or
+                    // 3. binding=refresh and is not the initial render
+                    !bindingAttrValue || bindingAttrValue === 'always' ||
+                    (bindingAttrValue === 'refresh' &&
+                      // bind.signals().get('FIRST_MUTATE') gives the timestamp (in ms) when mutation
+                      // occured, which is null for the initial render
+                      bind.signals().get('FIRST_MUTATE') !== null),
+                });
+              })
+              .then(() => {
+                return dict({__html: element.innerHTML}); // or outerHTML?
+              };
```

Comment:
nit: implicit return

```suggestion
              .then(() => dict({__html: element.innerHTML})); // or outerHTML?
```